core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -1407,21 +1407,24 @@ class WorkspaceService(protected val userInfo: UserInfo, val dataSource: SlickDa
       (dataAccess: DataAccess, workspaceContext: SlickWorkspaceContext, wdl: String, header: SubmissionValidationHeader, successes: Seq[SubmissionValidationEntityInputs], failures: Seq[SubmissionValidationEntityInputs], workflowFailureMode: Option[WorkflowFailureMode]) =>
         requireComputePermission(workspaceContext.workspace, dataAccess) {
           val submissionId: UUID = UUID.randomUUID()
+          val submissionEntityOpt = if(header.entityType.isEmpty) { None } else { Some(AttributeEntityReference(entityType = submissionRequest.entityType.get, entityName = submissionRequest.entityName.get)) }
 
           val workflows = successes map { entityInputs =>
+            val workflowEntityOpt = header.entityType.map(_ => AttributeEntityReference(entityType = header.entityType.get, entityName = entityInputs.entityName))
             Workflow(workflowId = None,
               status = WorkflowStatuses.Queued,
               statusLastChangedDate = DateTime.now,
-              workflowEntity = AttributeEntityReference(entityType = header.entityType, entityName = entityInputs.entityName),
+              workflowEntity = workflowEntityOpt,
               inputResolutions = entityInputs.inputResolutions
             )
           }
 
           val workflowFailures = failures map { entityInputs =>
+            val workflowEntityOpt = header.entityType.map(_ => AttributeEntityReference(entityType = header.entityType.get, entityName = entityInputs.entityName))
```

Comment:
`workflowEntityOpt` is defined once in `workflows` and again here in `workflowFailures`, maybe pull it out up top as something like `def workflowEntityOpt(entityName: String) = header.entityType.map(_ => AttributeEntityReference(entityType = header.entityType.get, entityName))`

Reply:
good catch :+1:

Reply:
oh, I can't -- they both uses `entityInputs` from their respective successes/failure maps